### PR TITLE
Mbarba/slurm2

### PR DESCRIPTION
**Pull request by MatBarba**
To add Slurm support, import make_resource method from prod-imported base runnable EGGeneric_conf in a new base runnable, then use it to define all resources.

### CODE REVIEW

File: lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  my $long = "24:00:00";
+
+  my %resources = (
+    %{$self->SUPER::resource_classes},
+    'small'    => $self->make_resource({"queue" => $queue, "memory" => 100, "time" => $short}),
+  );
+
+  # Additional names in the form xGB e.g. "2GB"
+  # From 2GB to 64GB
+  my @mems = (2, 4, 8, 12, 16, 32, 64);
```
Comment (vsitnik) [inline lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm:47]: unfortunately we need 128 and 256 sometimes

File: lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_genome_prepare_conf.pm [hunk 2]
```diff
@@ -206,7 +204,7 @@ sub pipeline_analyses {
       -language => 'python3',
       -analysis_capacity => 1,
       -failed_job_tolerance => 100,
-      -rc_name        => 'default',
+      -rc_name        => 'normal',
```
Comment (vsitnik) [inline lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_genome_prepare_conf.pm:207]: is it `normal` or `default`?

**Review by vsitnik — COMMENTED** [review]
thanks!

File: lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_genome_prepare_conf.pm (hunk 2 above)
Comment (MatBarba) [reply lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_genome_prepare_conf.pm:207]: default is for short quick jobs (<1h), I changed it to "normal" which can be up to 24h (needed in this case for some assemblies)

Comment (vsitnik) [reply lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_genome_prepare_conf.pm:207]: thanks!

File: lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm (hunk 1 above)
Comment (MatBarba) [inline lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm:47]:
```suggestion
  my @mems = (2, 4, 8, 12, 16, 32, 64, 128, 256);
```

Comment (ens-LCampbell) [reply lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm:47]: Not needed, bigmem queue specifies jobs that need >=300GB RAM

Comment (MatBarba) [reply lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm:47]: thanks, see the suggestion below then

**Review by ens-LCampbell — APPROVED** [review]
Ok with me for PR merge

File: lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm (hunk 1 above)
Comment (vsitnik) [reply lib/perl/Bio/EnsEMBL/Pipeline/PipeConfig/BRC4_base_conf.pm]: thanks!

**Review by vsitnik — APPROVED** [review]
thanks!

**Pull request merged**